File: nodejs/shadowxauth.js
const axios = require('axios');
const os = require('os');

/**
 * SHADOW X AUTH - Node.js SDK
 * GitHub: https://github.com/imranchowdhury3448-bit/SHADOW-X-AUTH-ALL-CS-FILE-
 * Discord: https://discord.gg/WwHrs5RVS
 */
class ShadowXAuth {
  constructor(ownerid, secret, version = '1.0') {
    this.ownerid = ownerid;
    this.secret = secret;
    this.version = version;
    this.apiUrl = 'https://shadowxauth.onrender.com/api/auth';
    this.isAuthenticated = false;
    this.message = '';
  }

  getHWID() {
    return `${os.hostname()}-${os.userInfo().username}`;
  }

  async login(username, password) {
    try {
      const res = await axios.post(`${this.apiUrl}/login`, {
        ownerid: this.ownerid,
        secret: this.secret,
        username,
        password,
        hwid: this.getHWID(),
        version: this.version
      }, { timeout: 35000 });

      if (res.data && res.data.success) {
        this.isAuthenticated = true;
        this.username = res.data.user.username;
        this.subscription = res.data.user.subscription;
        this.message = res.data.message || 'Authenticated';
        return true;
      }
      this.message = res.data.message || 'Authentication failed';
      return false;
    } catch (err) {
      this.message = err.response?.data?.message || err.message;
      return false;
    }
  }
}

module.exports = ShadowXAuth;

File: python/shadowxauth.py
import requests
import platform
import uuid

class ShadowXAuth:
    """
    SHADOW X AUTH - Python SDK
    GitHub: https://github.com/imranchowdhury3448-bit/SHADOW-X-AUTH-ALL-CS-FILE-
    Discord: https://discord.gg/WwHrs5RVS
    """
    API_URL = "https://shadowxauth.onrender.com/api/auth"

    def __init__(self, ownerid: str, secret: str, version: str = "1.0"):
        self.ownerid = ownerid
        self.secret = secret
        self.version = version
        self.is_authenticated = False
        self.username = ""
        self.subscription = ""
        self.expiry = ""
        self.message = ""

    def _get_hwid(self) -> str:
        return f"{platform.node()}-{uuid.getnode()}"

    def login(self, username: str, password: str) -> bool:
        payload = {
            "ownerid": self.ownerid,
            "secret": self.secret,
            "username": username,
            "password": password,
            "hwid": self._get_hwid(),
            "version": self.version
        }
        try:
            r = requests.post(f"{self.API_URL}/login", json=payload, timeout=35)
            data = r.json()
            if data.get("success"):
                self.is_authenticated = True
                user_info = data.get("user", {})
                self.username = user_info.get("username", username)
                self.subscription = user_info.get("subscription", "default")
                self.expiry = user_info.get("expiry", "")
                self.message = data.get("message", "Authenticated")
                return True
            self.message = data.get("message", "Login failed")
            return False
        except Exception as e:
            self.message = f"Connection error: {str(e)}"
            return False

    def register(self, username: str, password: str, email: str = "", license_key: str = "") -> bool:
        payload = {
            "ownerid": self.ownerid,
            "secret": self.secret,
            "username": username,
            "password": password,
            "email": email,
            "license_key": license_key,
            "hwid": self._get_hwid()
        }
        try:
            r = requests.post(f"{self.API_URL}/register", json=payload, timeout=35)
            data = r.json()
            self.message = data.get("message", "")
            return bool(data.get("success"))
        except Exception as e:
            self.message = f"Connection error: {str(e)}"
            return False

if __name__ == "__main__":
    auth = ShadowXAuth("YOUR_OWNERID", "YOUR_SECRET", "1.0")
    if auth.login("testuser", "testpass"):
        print(f"Welcome {auth.username}! Sub: {auth.subscription}")
    else:
        print(f"Error: {auth.message}")
